packages/astro/src/virtual-modules/middleware.ts

```diff
@@ -1 +1 @@
-export { defineMiddleware, sequence } from '../core/middleware/index.js';
+export { defineMiddleware, sequence, getOriginPathname } from '../core/middleware/index.js';
```

Comment:
I think `getOriginPathname` is something that users will use even outside of middleware. For example, endpoints have access to `APIContext`. 

Also, what do you think about exposing this function from `APIContext`/`Astro` globals e.g. `ctx.getOriginPathname()`/`Astro.getOriginPathname()`? Do you envision cases where users would need/want to handle actions in a frontmatter of an Astro file and/or endpoints?